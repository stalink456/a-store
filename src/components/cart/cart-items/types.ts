export enum OptionItems {
  Color = 'цвет',
  Size = 'размер',
  stickerName = 'номер стикера',
}
