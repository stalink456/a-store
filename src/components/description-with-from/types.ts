import { OptionShape } from "@alfalab/core-components/select";

export type OptionsTypes = Record<
  "colors" | "sizes" | "stickerNumbers",
  OptionShape
> | null;

export type DescriptionType = {
  id: number;
  preview: string;
  title: string;
  price: number;
  description: string;
  colors?: string[];
  sizes?: string[];
  stickerNumbers?: number[];
  availability?: boolean;
};
