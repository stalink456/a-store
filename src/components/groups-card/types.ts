import { GroupsType } from "store/types";

export type GroupsCardType = {
  items: GroupsType[];
};
