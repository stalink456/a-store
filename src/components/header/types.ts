export type HeaderContextType = {
  open?: boolean;
  handleOpenMenu?: () => void;
};
