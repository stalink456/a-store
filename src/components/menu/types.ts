export type MenuType = {
  open: boolean;
  handleOpenMenu: () => void;
};
