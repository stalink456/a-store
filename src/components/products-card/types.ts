export type ProductsCardType = {
  id: number;
  preview: string;
  title: string;
  subtitle?: string;
  price: number;
};
