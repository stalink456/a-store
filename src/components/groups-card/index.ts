export { GroupsCard } from './groups-card';
