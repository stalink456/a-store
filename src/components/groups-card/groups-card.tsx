import React from "react";
import { Space } from "@alfalab/core-components/space";
import { Typography } from "@alfalab/core-components/typography";
import { ProductsCard } from "components/products-card";
import { GroupsCardType } from "./types";

import style from "./groups-card.module.css";

export const GroupsCard: React.FC<GroupsCardType> = React.memo(({ items }) => {
  return (
    <div className={style.groups__card}>
      {items.map(({ title, description, products }, id) => (
        <Space direction="vertical" fullWidth key={id} size={0}>
          <Space
            className={style.groups__card_title}
            direction="vertical"
            fullWidth
          >
            <Typography.Title tag="h1" view="xlarge" weight="bold">
              {title}
            </Typography.Title>
          </Space>
          <Space
            className={style.groups__card_sub_title}
            direction="vertical"
            fullWidth
          >
            <Typography.Title tag="h6" view="small" weight="medium">
              {description}
            </Typography.Title>
          </Space>
          <Space
            className={style.groups__card__products}
            direction="horizontal"
            wrap
            size={0}
          >
            {products.map((value, id) => (
              <ProductsCard {...value} key={id} />
            ))}
          </Space>
        </Space>
      ))}
    </div>
  );
});
