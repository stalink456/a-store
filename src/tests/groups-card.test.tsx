import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Product from 'pages/product';
import { GroupsCard } from 'components/groups-card';
import { GroupsType } from 'store/types';
import * as loading from './mocks/get-loading-false.json';
import * as product from './mocks/get-product-200.json';
import * as groups from './mocks/get-groups-200.json';

const mockedLoading = loading['loading'];
const mockProduct = product['product'];
const mockedGroups = groups['groups'] as GroupsType[];

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('hooks/use-product', () => ({
  useProduct: () => ({
    product: mockProduct,
    isLoading: mockedLoading,
  }),
}));

describe('GroupsCard component', () => {
  it('should render GroupsCard component', async () => {
    const { baseElement } = render(<GroupsCard items={mockedGroups} />, {
      wrapper: MemoryRouter,
    });

    expect(baseElement).toMatchSnapshot();
  });

  it('should render GroupsCard component with title', async () => {
    render(<GroupsCard items={mockedGroups} />, {
      wrapper: MemoryRouter,
    });

    const result = await screen.findByText('Бархатные стикеры');
    expect(result).toBeInTheDocument();
  });

  it('should render GroupsCard component with a transition to Product page', async () => {
    render(
      <MemoryRouter>
        <Routes>
          <Route path='/' element={<GroupsCard items={mockedGroups} />} />
          <Route path='/product/:id' element={<Product />} />
        </Routes>
      </MemoryRouter>
    );

    fireEvent.click(screen.getByAltText('Футболка с бархатными стикерами'));

    fireEvent.click(
      screen.getByAltText(
        'http://qa-games.ru/astore/public/images/23597101.png'
      )
    );

    expect(
      await screen.findByText('Футболка с бархатными стикерами')
    ).toBeInTheDocument();
  });

  it("shouldn't render YourDesign component with non-existent image", async () => {
    render(<GroupsCard items={mockedGroups} />, {
      wrapper: MemoryRouter,
    });

    const image = screen.queryByAltText(
      'non-existent image'
    ) as HTMLImageElement;
    await waitFor(() => expect(image === null).toBeTruthy());
  });

  it("shouldn't render YourDesign component with non-existent description", async () => {
    render(<GroupsCard items={mockedGroups} />, {
      wrapper: MemoryRouter,
    });

    const result = screen.queryByAltText(
      'non-existent description'
    ) as HTMLImageElement;
    await waitFor(() => expect(result).not.toBeTruthy());
  });
});
