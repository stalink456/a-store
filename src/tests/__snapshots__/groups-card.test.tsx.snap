// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsCard component should render GroupsCard component 1`] = `
<body>
  <div>
    <div
      class="groups__card"
    >
      <div
        class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as"
      >
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
          style="margin-bottom: 0px;"
        >
          <div
            class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as groups__card_title"
          >
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
            >
              <h1
                class="typography__component_19zgr typography__styrene-xlarge_pxbut typography__bold_pxbut"
              >
                Бархатные стикеры
              </h1>
            </div>
          </div>
        </div>
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
          style="margin-bottom: 0px;"
        >
          <div
            class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as groups__card_sub_title"
          >
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
            >
              <h6
                class="typography__component_19zgr typography__styrene-small_pxbut typography__medium_pxbut"
              >
                Тактильный антистресс
              </h6>
            </div>
          </div>
        </div>
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
        >
          <div
            class="space__spaceContainer_nr3as space__start_nr3as groups__card__products"
            style="flex-wrap: wrap; margin-bottom: 0px;"
          >
            <div
              class="space__spaceItem_nr3as"
              style="padding-bottom: 0px;"
            >
              <div
                class="products_card"
              >
                <a
                  href="/product/6"
                >
                  <div
                    class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
                    style="flex-wrap: wrap; margin-bottom: 0px;"
                  >
                    <div
                      class="space__spaceItem_nr3as"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceContainer_nr3as space__start_nr3as"
                      >
                        <div
                          class="space__spaceItem_nr3as"
                        >
                          <div
                            class="skeleton__component_1bshy skeleton__animate_1bshy"
                          >
                            <img
                              alt="Футболка с бархатными стикерами"
                              class=""
                              src="http://qa-games.ru/astore/public/images/61646585.png"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="space__spaceItem_nr3as"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceContainer_nr3as space__start_nr3as"
                      >
                        <div
                          class="space__spaceItem_nr3as"
                        >
                          <h6
                            class="typography__component_19zgr typography__styrene-xsmall_pxbut typography__medium_pxbut"
                          >
                            Футболка с бархатными стикерами
                          </h6>
                        </div>
                      </div>
                    </div>
                    <div
                      class="space__spaceItem_nr3as"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceContainer_nr3as space__start_nr3as"
                      >
                        <div
                          class="space__spaceItem_nr3as"
                        >
                          <p
                            class="typography__paragraph_c16y5 typography__tertiary-inverted_hpb71 typography__primary-medium_c16y5 typography__bold_c16y5"
                            data-testid="subtitle-6"
                          >
                            Все варианты — внутри
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="space__spaceItem_nr3as"
                    >
                      <div
                        class="space__spaceContainer_nr3as space__start_nr3as"
                      >
                        <div
                          class="space__spaceItem_nr3as"
                        >
                          <p
                            class="typography__paragraph_c16y5 typography__primary-large_c16y5 typography__bold_c16y5"
                            data-testid="price-6"
                          >
                            1799
                             ₽
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
